Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs

```diff
@@ -0,0 +1,179 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS, 
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+#endregion
+
+using Amdocs.Ginger.Common;
+using Amdocs.Ginger.CoreNET.RunLib.DynamicExecutionLib;
+using Ginger.ExecuterService.Contracts.V1.ExecutionConfiguration;
+using Ginger.Run;
+using Ginger.SolutionGeneral;
+using System;
+
+namespace Amdocs.Ginger.CoreNET.RunLib.CLILib
+{
+    public class CLIDynamicFile : ICLI
+    {
+        public enum eFileType { XML,JSON}
+
+        bool ICLI.IsFileBasedConfig { get { return true; } }
+
+        public string Verb
+        {
+            get
+            {
+                return DynamicOptions.Verb;
+            }
+        }
+
+        string ICLI.FileExtension
+        {
+            get
+            {
+                if (FileType == eFileType.JSON)
+                {
+                    return "json";
+                }
+                else
+                {
+                    return "xml";
+                }
+            }
+        }
+
+        public eFileType FileType { get; set; }
+
+        public CLIDynamicFile(eFileType fileType)
+        {
+            FileType = fileType;
+        }
+
+        public string CreateConfigurationsContent(Solution solution, RunsetExecutor runsetExecutor, CLIHelper cliHelper)
+        {
+            if (FileType == eFileType.JSON)
+            {
+                string json = DynamicExecutionManager.CreateDynamicRunSetJSON(solution, runsetExecutor, cliHelper);
+                return json;
+            }
+            else
+            {
+                string xml = DynamicExecutionManager.CreateDynamicRunSetXML(solution, runsetExecutor, cliHelper);
+                return xml;
+            }
+        }
+
+        public void LoadGeneralConfigurations(string content, CLIHelper cliHelper)
+        {
+            if(FileType == eFileType.JSON)
+            {
+                //Dynamic JSON
+                GingerExecConfig exeConfiguration = DynamicExecutionManager.LoadDynamicExecutionFromJSON(content);
+                if (exeConfiguration.SolutionScmDetails != null)
+                {
+                    cliHelper.SetSourceControlType(exeConfiguration.SolutionScmDetails.SCMType.ToString());
+                    cliHelper.SetSourceControlURL(exeConfiguration.SolutionScmDetails.SolutionRepositoryUrl);
+                    cliHelper.SetSourceControlUser(exeConfiguration.SolutionScmDetails.User);
+                    cliHelper.SetSourceControlPassword(exeConfiguration.SolutionScmDetails.Password);
+                    cliHelper.PasswordEncrypted(exeConfiguration.SolutionScmDetails.PasswordEncrypted.ToString());
+                    if (string.IsNullOrEmpty(exeConfiguration.SolutionScmDetails.ProxyServer) == false)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4812633)